src/buffer.h

```diff
@@ -154,4 +175,50 @@ mpd_buffer_consume(struct mpd_buffer *buffer, size_t nbytes)
 	buffer->read += nbytes;
 }
 
+/**
+ * Makes at least min_size bytes available for writing data.
+ * In other words, mpd_buffer_room() will be >= min_size if called after this
+ * function.
+ */
+static inline bool
+mpd_buffer_make_room(struct mpd_buffer *buffer, size_t min_avail_len)
+{
+	size_t newsize;
+
+	if (mpd_buffer_room(buffer) >= min_avail_len)
+		return true;
+
+	newsize = buffer->data_len * 2;
+	while (newsize < min_avail_len)
+		newsize *= 2;
+
+	/* empty buffer */
+	if (mpd_buffer_room(buffer) == buffer->data_len) {
+		free(buffer->data);
+		buffer->data = malloc(newsize);
+		if (buffer->data == NULL)
+			return false;
+		memset(buffer->data, 0, newsize);
+	} else {
+		buffer->data = realloc(buffer->data, newsize);
+		if (buffer->data == NULL)
+			return false;
+
+		/* clear region not committed and new region */
+		memset(mpd_buffer_write(buffer), 0,
+		       newsize - (buffer->data_len - mpd_buffer_room(buffer)));
```

Comment:
Why?

Reply:
Remove stale (non committed) data from the buffer

Reply:
I don't understand.

Reply:
This is to keep the data in a known state.
We always initialize it to zero.
If the user called `mpd_buffer_make_room` (either directly or indirectly), it may be that it wrote some data and realized that it did not have enough space for it (function `quote_vargs`); thus, that area is reverted to the zero state. Otherwise, if no non-committed data is present, this memset will do nothing (last parameter is zero).

Reply:
But what's the point of that? Why is "zero state" any more special than "random uncommitted data"? Zero is an arbitrary value just like any other.
I don't like that we discuss this here, in a PR and in a commit that has nothing to do with this. You're sneaking in an unrelated change.

Reply:
okay, I will remove the code related to the known state